packages/in-app-notification-manager/test/api/getNotifications.test.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi, beforeEach, afterEach } from "vitest";
+import { generateId } from "pagopa-interop-models";
+import { generateToken } from "pagopa-interop-commons-test";
+import { AuthRole, authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { api, inAppNotificationService } from "../vitest.api.setup.js";
+
+describe("API GET /notifications", () => {
+  const makeRequest = async (
+    token: string,
+    queryParams: Record<string, string> = {
+      limit: "10",
+      offset: "0",
+    }
+  ) =>
+    request(api)
+      .get("/notifications")
+      .query(queryParams)
+      .set("Authorization", `Bearer ${token}`)
+      .set("X-Correlation-Id", generateId())
+      .send();
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    inAppNotificationService.getNotifications = vi.fn().mockResolvedValue({
+      results: [],
+      totalCount: 0,
```

Comment:
maybe we can test with a non-empty array to check also the mapping in the router?